src/scripts/main.js

```diff
@@ -1,3 +1,212 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const sortBy = document.querySelector('tr');
+const tBody = document.querySelector('tbody');
+const rows = tBody.querySelectorAll('tr');
+const minAge = 18;
+const maxAge = 90;
+
+body.insertAdjacentHTML('beforeend', `
+<form class="new-employee-form">
+<label>Name: <input name="name" type="text" data-qa="name" required></label>
+<label>Position:
+  <input name="position" type="text" data-qa="position" required>
+</label>
+<label>Office:
+  <select name="office" data-qa="office" required>
+    <option value="Tokyo">Tokyo</option>
+    <option value="Singapore">Singapore</option>
+    <option value="London">London</option>
+    <option value="New York">New York/option>
+    <option value="Edinburgh">Edinburgh</option>
+    <option value="San Francisco">San Francisco</option>
+  </select></label>
+<label>Age: <input name="age" type="number" data-qa="age" required></label>
+<label>Salary:
+<input name="salary" type="number" data-qa="salary" required>
+</label>
+<button type="submit">Save to table</button>
+</form>
+`);
+
+const pushNotification = (title, description, type) => {
+  body.insertAdjacentHTML('beforeend', `
+<div class="notification ${type}" data-qa="notification">
+<h2 class="title">
+  ${title}
+</h2>
+
+<p>
+  ${description}
+</p>
+</div>
+`);
+
+  const notif = document.querySelectorAll('.notification');
+
+  setTimeout(() => {
+    notif.forEach(message => message.remove());
+  }, 3000);
+};
+
+const form = document.querySelector('form');
+const button = form.querySelector('button');
+const inputName = form.querySelector('input[data-qa="name"]');
+const inputAge = form.querySelector('input[data-qa="age"]');
+const inputPosition = form.querySelector('input[data-qa="position"]');
+const inputSalary = form.querySelector('input[data-qa="salary"]');
+const inputOffice = form.querySelector('select[data-qa="office"]');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault(); e.target.reset();
+});
+
+button.addEventListener('click', e => {
+  const nameLength = inputName.value.length;
+  const age = inputAge.value;
+
+  if (nameLength < 4 || age < minAge || age > maxAge) {
```

Comment:
```suggestion
  const nameLength = inputName.value.length;
  const age = inputAge.value;
  const isErrorNotification = nameLength < 4 || age < minAge || age > maxAge;
  
  if (isErrorNotification) {
```
making variable for long condition make code more readable and understanddable